ndmg/utils/cloud_utils.py

```diff
@@ -123,44 +123,55 @@ def s3_get_data(bucket, remote, local, public=False, force=False):
     remote : str
         The path to the data on your S3 bucket. The data will be
         downloaded to the provided bids_dir on your machine.
-    local : str
-        Local input directory where you want the files copied to
+    local : list
+        Local input directory where you want the files copied to and subject/session info [input, sub-#/ses-#]
+    info : str, optional
+        Relevant subject and session information in the form of sub-#/ses-#
     public : bool, optional
         Whether or not the data you are accessing is public, if False s3 credentials are used, by default False
     force : bool, optional
         Whether to overwrite the local directory containing the s3 files if it already exists, by default False
     """
 
-    # TODO : use boto3 for this
-    if os.path.exists(local) and not force:
-        print("Local directory already exists. Not pulling s3 data.")
-        return  # TODO: make sure this doesn't append None a bunch of times to a list in a loop on this function
+    if os.path.exists(os.path.join(local,info)) and not force:
+        if os.listdir(os.path.join(local,info)):
+            print(f'Local directory: {os.path.join(local,info)} already exists. Not pulling s3 data. Delete contents to re-download data.')
+            return  # TODO: make sure this doesn't append None a bunch of times to a list in a loop on this function
+        else:
+            pass
+    
     if not public:
         # get client with credentials if they exist
         try:
             client = s3_client(service="s3")
         except:
             client = boto3.client("s3")
+    else:
+        client = boto3.client("s3")
+    
+    # check that bucket exists
+    bkts = [bk["Name"] for bk in client.list_buckets()["Buckets"]]
+    if bucket not in bkts:
+        sys.exit(
+            "Error: could not locate bucket. Available buckets: " + ", ".join(bkts)
+        )
 
-        bkts = [bk["Name"] for bk in client.list_buckets()["Buckets"]]
-        if bucket not in bkts:
-            sys.exit(
-                "Error: could not locate bucket. Available buckets: " + ", ".join(bkts)
-            )
+    # go through all folders inside of remote directory and download relevant files
+    s3 = boto3.resource('s3')
+    buck = s3.Bucket(bucket)
+    for obj in buck.objects.filter(Prefix=f'{remote}'):
+        bdir,data = os.path.split(obj.key)
+        print(os.path.join(local,*bdir.split('/')[-3:]))
+        # Make directory for data if it doesn't exist
+        if not os.path.exists(os.path.join(local,*bdir.split('/')[-3:])):
+            os.makedirs(os.path.join(local,*bdir.split('/')[-3:]))
+        print(f'Downloading {bdir}/{data} from {bucket} s3 bucket...')
```

Comment:
`os.path.join(local,*bdir.split('/')[-3:]` should just be defined once as a well-named variable at the top